src/data-types/wrappers.h

```diff
@@ -0,0 +1,326 @@
+#pragma once
+
+#include <errno.h>
+#include <stdarg.h>
+
+#ifdef EOF // stdio.h
+
+static inline FILE * Fopen(const char *filename, const char *mode)
+{
+    FILE * f;
+
+    do {
+        f = fopen(filename, mode);
+    } while (f == NULL && errno == EINTR);
+
+    return f;
+}
+#define fopen >@<
```

Comment:
On Thu, Mar 05, 2015 at 01:39:06PM -0800, Hoà V. DINH wrote:
> > +#define fopen >@<
> I'm not sure to understand this syntax. Is it part of C standard?

The reason for this is that it generates a syntax error for sure. Why?
Because if you're using wrappers you shouldn't forget to use them in one
line. So in a file where you're using wrappers you'll get syntax errors
if you're forgetting things.

This is for code maintenance reasons.

Yours,
VB.
-- 
Volker Birk
Oberer Graben 4, 8400 Winterthur, Schweiz
mailto:vb@dingens.org  http://fdik.org